packages/vm/tests/api/customPrecompiles.spec.ts

```diff
@@ -0,0 +1,120 @@
+import tape from 'tape'
+import VM from '../../src'
+import { Address } from 'ethereumjs-util'
+import { PrecompileInput } from '../../src/evm/precompiles'
+import { ExecResult } from '../../src/evm/evm'
+
+const sender = new Address(Buffer.from('44'.repeat(20), 'hex'))
+const newPrecompile = new Address(Buffer.from('ff'.repeat(20), 'hex'))
+const shaAddress = new Address(Buffer.from('0000000000000000000000000000000000000002', 'hex'))
+const expectedReturn = Buffer.from('1337', 'hex')
+const expectedGas = BigInt(10)
+
+// TODO if `opts` is removed (so function has no arguments) then tests fail!? Why?
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function customPrecompile(opts: PrecompileInput): ExecResult {
```

Comment:
just looked into this, I believe because of PrecompileFunc's definition

```typescript
export interface PrecompileFunc {
  (opts: PrecompileInput): Promise<ExecResult> | ExecResult
}
```

if the number of params don't match when the function is called (here: one, `opts`) then it would error when calling the method.